РАБОЧАЯ ВЕТКА

kreshvan opened the pull request.

kreshvan committed:
РАБОЧАЯ ВЕТКА

kreshvan committed:
ошибка

kreshvan committed:
дописал Тесты

AleksandrZhukovJava commented on src/main/java/groupId/ru/hogwarts/school/controller/StudentController.java:
```diff
@@ -18,10 +19,23 @@
+    private final StudentServiceImpl studentServiceImpl;
+
+
+    @GetMapping("/sum-students-by-id")
```
Ендпоинт должен вернуть просто число, сколько конкретно студентов на всех факультетах, а не дто
Название надо поменять, тут нет привязки к id (не должно быть)

AleksandrZhukovJava commented on src/main/java/groupId/ru/hogwarts/school/controller/StudentRepository.java:
```diff
@@ -3,16 +3,21 @@
 
 import static java.lang.Integer.MAX_VALUE;
-
+@Repository
```
Не обязательная аннотация для JpaRepository

AleksandrZhukovJava commented on src/main/java/groupId/ru/hogwarts/school/controller/StudentRepository.java:
```diff
@@ -3,16 +3,21 @@
     //int minNumber = 0;
     //int maxNumber = MAX_VALUE;
     List<Student> findByAgeBetween(int min, int max);
+    @Query(value = "SELECT category,SUM (id) as id From students GROUP BY category",nativeQuery = true)
```
почти верная Query просто не нужна категория, только sum(id) без группировки, и метод возвращает число

kreshvan committed:
ошибка при запуске

AleksandrZhukovJava commented on src/main/java/groupId/ru/hogwarts/school/controller/StudentController.java:
```diff
@@ -0,0 +1,125 @@
+    if (id>=0|| file!=null){
+        return new ResponseEntity<>("Аватарка успешно загружена", HttpStatus.OK);
+    }
+        return new ResponseEntity<>("Ошибка при загрузке аватарки", HttpStatus.INTERNAL_SERVER_ERROR);
```
500 код ответа не для бизнесовых ошибок, специально мы никогда не используем 5хх тип, только 4хх ые

kreshvan committed:
SQL and JOIN ЗАПРОСЫ

kreshvan committed:
SQL ЗАПРОСЫ INDEX INDEX INDEX

kreshvan committed:
ПРОФИЛИ ПОРТЫ application.properties

AleksandrZhukovJava commented on src/main/java/groupId/ru/hogwarts/school/controller/InfoController.java:
```diff
@@ -0,0 +1,31 @@
+@RequestMapping("/port")
+public class InfoController {
+
+@Value("${server.port0},${server.port1},${server.port2}")
```
Это не заработает, нужно прописать для каждой переменной отдельно

AleksandrZhukovJava commented on src/main/java/groupId/ru/hogwarts/school/controller/InfoController.java:
```diff
@@ -0,0 +1,31 @@
+    private int port2;
+
+@GetMapping("/0")
+    public int getPort0(){
```
по заданию путь другой, от тех задания отходить нельзя, на него на настоящей работе будет опираться фронт, в итоге этот ендпоинт просто не будет работать в приложении

AleksandrZhukovJava commented on src/main/java/groupId/ru/hogwarts/school/service/StudentServiceImpl.java:
```diff
@@ -53,6 +88,11 @@ public List<Student> findAllPage(Integer pageNumber, Integer pageSize) {
 
 
 
+    @Autowired
```
Почему @Autowired? Так мы внедряем бины, а тут просто метод

kreshvan committed:
# 4.5 Параллельные стримы
ЦЕль сегодняшней домашней работы — написать несколько модификаторов для структур данных, используя Stream API.

kreshvan committed:
ПОТОКИ
